docs/langref/src/concepts/purity.md

```diff
@@ -6,22 +6,25 @@ automatically discovered during property evaluation. The property stores the
 result of the evaluation. When a property changes, all dependent properties get
 notified, so that the next time their value is read, their binding is re-evaluated.
 
-For this reactive system to work well, bindings evaluation should be "pure":
-Evaluating a property shouldn't have side-effects. Side effects are
-problematic because it's not always clear when they will happen: Lazy evaluation
-may change their order or affect whether they happen at all. In addition, changes
-to properties during their binding evaluation due to a side-effect
-may result in unexpected behavior.
+For any reactive system to work well, bindings evaluation should be "pure":
+Evaluating a property shouldn't have side-effects nor change
+any observable state but the property itself. Side effects are problematic
```

Comment:
Let's use the idiomatic "neither" for the conjunction then.

```suggestion
Evaluating a property should have neither side-effects nor change
any observable state but the property itself. Side effects are problematic
```

That said, I prefer the original way it was written. A change to observable state other than the property itself **is** a side-effect after all, isn't it?

Reply:
I'm inclined to agree, but "neither ... nor" with two sides that are not actually mutually exclusive is more confusing IMO than helping. How about two sentences?

"Evaluating a property shouldn't have side-effects. An example of a side-effect is when a binding expression not only computes the value of the property it's associated with, but also assigns new values to other properties."